check_pgactivity

```diff
@@ -5680,76 +5687,147 @@ sub check_replication_slots {
+# ----------+-----------+------------- -----+---------------
+# mysub3    | logical   |                -1 |             1
+# mysub     | logical   |                -1 |             1
     );
 
     @hosts = @{ parse_hosts %args };
 
-    is_compat $hosts[0], 'replication_slots', $PG_VERSION_94 or exit 1;
-
     pod2usage(
         -message => 'FATAL: you must give only one host with service "replication_slots".',
         -exitval => 127
     ) if @hosts != 1;
 
-    if ( defined $args{'warning'} and defined $args{'critical'} ) {
-        # warning and critical must be raw
-        pod2usage(
-            -message => "FATAL: critical and warning thresholds only accept raw numbers.",
-            -exitval => 127
-        ) unless $args{'warning'}  =~ m/^([0-9]+)$/
-            and  $args{'critical'} =~ m/^([0-9]+)$/;
-    }
+    is_compat $hosts[0], 'replication_slots', $PG_VERSION_94 or exit 1;
 
-    @rs = @{ query_ver( $hosts[0], %queries ) };
+    if ( defined $args{'warning'} ) {
+        my $threshods_re = qr/(wal|replslot)\s*=\s*(?:(\d+))/i;
 
-SLOTS_LOOP: foreach my $row (@rs) {
-        push @perfdata => [ $row->[0], $row->[1] ];
+        if ($args{'warning'} =~ m/^$threshods_re(\s*,\s*$threshods_re)*$/
+            and $args{'critical'} =~ m/^$threshods_re(\s*,\s*$threshods_re)*$/) {
 
-        if ( defined $args{'warning'} ) {
+              while ( $args{'warning'} =~ /$threshods_re/g ) {
+                  my ($threshold, $value) = ($1, $2);
+                    $warn{$threshold.'_files'} = $value;
+              }
+              while ( $args{'critical'} =~ /$threshods_re/g ) {
+                  my ($threshold, $value) = ($1, $2);
+                    $crit{$threshold.'_files'} = $value;
+              }
 
-            if ( $row->[1] >= $args{'critical'} ) {
-                push @msg_crit => "$row->[0]: $row->[1] WAL";
-                next SLOTS_LOOP;
             }
+          # For backward compatibility
+          elsif ($args{'warning'}  =~ m/^([0-9]+)$/
+                 and $args{'critical'} =~ m/^([0-9]+)$/) {
+                   $warn{wal_files} = $args{'warning'};
+                   $crit{wal_files} = $args{'critical'};
+                 }
+          else {
+            pod2usage(
+                -message => "FATAL: critical and warning thresholds only accept:\n"
+                    . "- raw numbers for backward compatibility to set wal threshold.\n"
+                    . "- a list 'wal=value' and/or 'replslot=value' separated by comma.\n"
+                    . "See documentation for more information.",
+                -exitval => 127
+            )
+          }
 
-            if ( $row->[1] >= $args{'warning'} ) {
-                push @msg_warn => "$row->[0]: $row->[1] WAL";
-            }
+        dprint ('replication_slots_warn: '. Dumper(%warn));
+        dprint ('replication_slots_crit: '. Dumper(%crit));
+
+    }
+
+    @perf_wal_limits = ( $warn{wal_files}, $crit{wal_files} ) if defined $warn{wal_files};
```

Comment:
add `or defined $crit{wal_files};` 
or a `-c 'wal=99` will not appear in the perfdata